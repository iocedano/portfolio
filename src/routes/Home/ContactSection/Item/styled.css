.item-container-img {
  width: 50px;
}
