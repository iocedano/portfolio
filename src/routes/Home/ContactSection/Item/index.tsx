import React, { useState, useEffect, memo } from 'react';
import PropType, { InferProps } from 'prop-types';
import './styled.css';

const ItemPropTypes = {
  type: PropType.oneOf(['github', 'linkin']).isRequired,
  url: PropType.string.isRequired
};

export type ItemType = InferProps<typeof ItemPropTypes>;

const Item = ({ type, url } : ItemType) => {
  const [img, setImg] = useState('');

  useEffect(() => {
    import(`../../../../assets/icons/${type}.svg`).then((image) => {
      setImg(image.default);
    });
  }, [type]);

  return (
    <li className="item-container">
      <a href={url} target="_blank" rel="noopener noreferrer">
        <img className="item-container-img" src={img} alt={`contact me ${type}`}/>
      </a>
    </li>
  );
};

export default memo(Item);
