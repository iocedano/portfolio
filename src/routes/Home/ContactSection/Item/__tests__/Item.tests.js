import React from 'react';
import { render, act } from '@testing-library/react';
import Item from '..';

describe('Home/ContactSection/Item', () => {
  it('should render', async () => {
    const promise = Promise.resolve();
    const { container } = render(<Item type="github" url="https://localhost"/>);

    expect(container.firstChild).toMatchSnapshot();
    await act(() => promise);
  });
});
