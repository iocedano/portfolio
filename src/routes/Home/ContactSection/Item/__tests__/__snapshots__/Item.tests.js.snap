// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home/ContactSection/Item should render 1`] = `
<li
  class="item-container"
>
  <a
    href="https://localhost"
    rel="noopener noreferrer"
    target="_blank"
  >
    <img
      alt="contact me github"
      class="item-container-img"
      src=""
    />
  </a>
</li>
`;
